x-pack/platform/plugins/shared/ml/public/embeddables/anomaly_swimlane/anomaly_swimlane_setup_flyout.tsx

```diff
@@ -7,67 +7,42 @@
 
 import type { CoreStart } from '@kbn/core/public';
 import { KibanaContextProvider } from '@kbn/kibana-react-plugin/public';
-import { tracksOverlays } from '@kbn/presentation-util';
-import { toMountPoint } from '@kbn/react-kibana-mount';
 import React from 'react';
+import type { MlDependencies } from '../../application/app';
 import type { AnomalySwimLaneEmbeddableState, AnomalySwimlaneEmbeddableUserInput } from '..';
 import { HttpService } from '../../application/services/http_service';
 import { jobsApiProvider } from '../../application/services/ml_api_service/jobs';
 import { AnomalySwimlaneInitializer } from './anomaly_swimlane_initializer';
+import type { MlStartDependencies } from '../../plugin';
+import { getMlGlobalServices } from '../../application/util/get_services';
 
-export async function resolveAnomalySwimlaneUserInput(
-  coreStart: CoreStart,
-  parentApi: unknown,
-  focusedPanelId: string,
-  input?: Partial<AnomalySwimLaneEmbeddableState>
-): Promise<AnomalySwimlaneEmbeddableUserInput> {
-  const { http, overlays, ...startServices } = coreStart;
-
-  const overlayTracker = tracksOverlays(parentApi) ? parentApi : undefined;
-
-  return new Promise(async (resolve, reject) => {
-    try {
-      const adJobsApiService = jobsApiProvider(new HttpService(http));
-
-      const flyoutSession = overlays.openFlyout(
-        toMountPoint(
-          <KibanaContextProvider services={{ ...coreStart }}>
-            <AnomalySwimlaneInitializer
-              adJobsApiService={adJobsApiService}
-              initialInput={input}
-              onCreate={(explicitInput) => {
-                resolve(explicitInput);
-                flyoutSession.close();
-                overlayTracker?.clearOverlays();
-              }}
-              onCancel={() => {
-                reject();
-                flyoutSession.close();
-                overlayTracker?.clearOverlays();
-              }}
-            />
-          </KibanaContextProvider>,
-          startServices
-        ),
-        {
-          type: 'push',
-          ownFocus: true,
-          size: 's',
-          onClose: () => {
-            reject();
-            flyoutSession.close();
-            overlayTracker?.clearOverlays();
-          },
-        }
-      );
-
-      if (tracksOverlays(parentApi)) {
-        parentApi.openOverlay(flyoutSession, {
-          focusedPanelId,
-        });
-      }
-    } catch (error) {
-      reject(error);
-    }
-  });
+export function ResolveAnomalySwimlaneUserInput({
+  coreStart,
+  pluginStart,
+  onConfirm,
+  onClose,
+  input,
+}: {
+  coreStart: CoreStart;
+  pluginStart: MlDependencies | MlStartDependencies;
+  onConfirm: (state: AnomalySwimlaneEmbeddableUserInput) => void;
+  onClose: () => void;
```

Comment:
please rename this to onCancel to keep it user-domain specific. For the onConfirm callback you have to remove the `onClose` part.